api/job/job.go

```diff
@@ -29,27 +32,89 @@ func newHttpJob(client *smclient.Client) httpJob {
 	return httpJob{client: client}
 }
 
-// GetJob gets a job related info
-func (h httpJob) GetJob(projectID string, translationJobUID string) (GetJobResponse, error) {
+// Get gets a job related info
+func (h httpJob) Get(projectID string, translationJobUID string) (GetJobResponse, error) {
 	url := jobBasePath + projectID + "/jobs/" + translationJobUID
 	var response getJobResponse
 	rawMessage, code, err := h.client.Get(url, nil)
 	if err != nil {
 		return GetJobResponse{}, err
 	}
+	defer func() {
+		if err := rawMessage.Close(); err != nil {
+			h.client.Logger.Debugf("failed to close response body: %v", err)
+		}
+	}()
+	body, err := io.ReadAll(rawMessage)
+	if err != nil {
+		return GetJobResponse{}, fmt.Errorf("failed to read response body: %w", err)
+	}
 	if code != 200 {
-		body, _ := io.ReadAll(rawMessage)
 		h.client.Logger.Debugf("response body: %s\n", body)
 		return GetJobResponse{}, fmt.Errorf("unexpected response code: %d", code)
 	}
+	if err := json.Unmarshal(body, &response); err != nil {
+		return GetJobResponse{}, fmt.Errorf("failed to unmarshal response: %w", err)
+	}
+	return toGetJobResponse(response), nil
+}
+
+// GetAllByName gets all jobs of a project by name
+func (h httpJob) GetAllByName(projectID, name string) ([]GetJobResponse, error) {
```

Comment:
Could you please research how to add validation for input parameters? An example for this specific case: we must check that parameters are not empty and `projectID` follows specific patterns.

P.S. Well, I do not want to broaden the scope of this ticket/PR. Let's make a Jira ticket for research on how to use a declarative approach for validation (rather than imerative).